### DISCUSSION TITLE
Microsoft Teams Notifications

### DISCUSSION THREAD

**Discussion by JustSaX** [FelixKratz/SketchyBar#514] [Q&A]
Was someone able to display Microsoft Teams notifications in the status bar? I would like to have the person/chat in the status bar who send me a Teams message. I don't want the message content. I just want to know who is the sender or in which chat it was sent.

Thanks!

**Comment by marconetto** [comment]
@JustSaX, have you made any progress on adding Microsoft Teams Notifications in SketchyBar?

**Comment by bens-insurgence** [comment]
I was able to get it working doing this.

`#!/bin/bash

# Check if Microsoft Teams is running
TEAMS_RUNNING=$(pgrep -x "MSTeams" > /dev/null && echo "true" || echo "false")

if [ "$TEAMS_RUNNING" = "false" ]; then
  sketchybar --set $NAME \
             label="" \
             icon.color=0xff494d64 \
             background.border_color=0xff2a2c3a \
             drawing=on
  exit 0
fi

# Get notification count from Dock badge
NOTIFICATION_COUNT=$(osascript -e '
tell application "System Events"
  tell UI element "Microsoft Teams" of list 1 of process "Dock"
    try
      set badgeValue to value of attribute "AXStatusLabel"
      if badgeValue is not missing value then
        return badgeValue
      end if
    end try
  end tell
end tell
return "0"' 2>/dev/null)

# Clean up the notification count (remove non-numeric characters and whitespace)
NOTIFICATION_COUNT=$(echo "$NOTIFICATION_COUNT" | tr -cd '0-9')

# Default to 0 if empty
if [ -z "$NOTIFICATION_COUNT" ]; then
  NOTIFICATION_COUNT=0
fi

# Update the widget
if [ "$NOTIFICATION_COUNT" -gt 0 ]; then
  sketchybar --set $NAME \
             label="$NOTIFICATION_COUNT" \
             icon.color=0xfff7768e \
             background.border_color=0xfff7768e \
             drawing=on
else
  sketchybar --set $NAME \
             label="" \
             icon.color=0xff7aa2f7 \
             background.border_color=0xff2a2c3a \
             drawing=on
fi
`

**Comment by marconetto** [comment]
thanks @JustSaX. I’ll give it a go in the near future!

**Comment by DarthImriel** [comment]
I have a Teams icon (using NerdFonts) in my bar when Teams is open; it's white if I have no notifications and turns red if there are notifications.

You'll either need a colors definition file to source, define colors at the top of the script, or replace the color variables with hex codes.

<details>
<summary>Plugin script</summary>

```
#!/usr/bin/env sh
source "$HOME/.config/sketchybar/colors.sh"

STATUS_LABEL=$(lsappinfo info -only StatusLabel "Microsoft Teams")
ICON="󰊻"

if [[ $STATUS_LABEL == *"NULL"* ]]; then
    ICON_COLOR="$WHITE"
elif [[ $STATUS_LABEL =~ \"label\"=\"?([^\"}[:space:]]*)\"? ]]; then
    LABEL="${BASH_REMATCH[1]}"

    if [[ $LABEL == "" ]]; then
        ICON_COLOR="$WHITE"
    elif [[ $LABEL == "•" ]]; then
        ICON_COLOR="$BLUE"
    elif [[ $LABEL =~ ^[0-9]+$ ]]; then
        ICON_COLOR="$RED"
    else
        ICON_COLOR="$WHITE"
    fi
else
    exit 0
fi

sketchybar --set $NAME icon=$ICON icon.color=${ICON_COLOR}
```

</details>


<details> 

<summary>Item</summary> 

```
sketchybar  --add   item teams right \
            --set   teams \
                    update_freq=5 \
                    script="$PLUGIN_DIR/teams.sh" \
                    icon.font.size=18 \
           --subscribe teams system_woke
```

</details>
